File: modules/org.openhs.core.commons/src/org/openhs/core/commons/WeatherData.java
/**
* @name		WeatherData.java 11/18/2017
* @author	Michal Valny
* @version	1.0
* @description 	Weather data...
* 
*/

package org.openhs.core.commons;

import java.util.Date;

public class WeatherData {
	
	public boolean validity = false;
	
	//Date date = new Date ();
	
	public String location = "Brno";
	
	public float temp;
	public float tempMin;
	public float tempMax;
	public float hum;
    public float pressure;
    public float cloudsPercent;        
    public float windSpeed;
    public float windDegree;    
    public float rain;
    public float snow;	
    
    public int weatherSymbol = 0; //0: not set, 1:Sunny, 2:Sunny with Cloud, 3: Cloudy, 4: Cloudy+Rain, 5:Cloudy+Storm, 6: Cloudy+Snow  
    
    public WeatherData () {

    }        
    
	public String toString() {
		
		String out = "";
		
		out = out + "\n validity: " + validity;
		out = out + "\n temp: " + temp;
		out = out + "\n hum: " + hum;

		return out;
	}    
}

File: web/InfoStation/src/app/OhsWeather/WeatherData.ts


export class WeatherData {
    public validity                 = false; // content of the forecast is valid
    public location                 = '';
    public temp                     = 0.0;
    public tempMin                  = 0.0;
    public tempMax                  = 0.0;
    public hum                      = 0.0;
    public cloudsPercent            = 0.0;
    public windSpeed                = 0.0;
    public windDegree               = 0.0;
    public rain                     = 0.0;
    public snow                     = 0.0;
    public weatherSymbol            = 0.0; // 0: not set, 1:Sunny, 2:Sunny with Cloud, 3: Cloudy, 4: Cloudy+Rain, 5:Cloudy+Storm, 6: Cloudy+Snow 

    constructor() {

    }

    fillFromJSON(json: any) {

        for (var propName in json) {
            this[propName] = json[propName];
        }
    }
}